src/metabase/api/user.clj

```diff
@@ -68,13 +68,15 @@
         last_name       (or last_name prev_last_name)]
     (when (or (not= first_name prev_first_name)
               (not= last_name prev_last_name))
-      [first_name last_name])))
+      [(when-not (= first_name :explicitly-nil) first_name)
+       (when-not (= last_name :explicitly-nil) last_name)])))
 
```

Comment:
I have to differentiate when the endpoint explicitly sets first/last name to 'nil' vs. just not setting the value at all. Otherwise, we may accidentally set a name to 'nil' when we don't mean to. For example, if `PUT user/:id` is used to update `:first_name` but NOT `:last_name`, we would expect the existing last name value to remain unchanged. If we use the endpoint to explicitly set either name to nil, we do want those changes to flow through.

Reply:
I don't see any tests for `:explicitly-nil` to verify that it actually does what you want... I'm also a little confused as to how it's supposed to work because the first name and last name are **strings** so no matter what they'll never equal a **keyword**. 

Either way I'm not sure we really need this. This is the way we usually do this sort of thing elsewhere in the codebase:

- For `POST` endpoints (to insert a new row) if the key is not passed in at all or if the key is passed in with a value of `nil` then the value should be `nil`
- For `PUT` endpoints (to update a record):
  - If the key is not passed in at all, don't update that column
  - If the key is passed in as `nil`, then set the value of that column to `nil`

Reply:
One more suggestion: it's not clear to me whether you intended to have the frontend pass in the `:explicitly-nil` or it's just something that you're using as a flag internally in the backend. If it's the latter I would suggest using a namespaced keyword to make it clearer that it's "private" to this namespace, e.g. `::explicitly-nil`, as opposed to something used elsewhere in the codebase. If it's meant to be passed in by the frontend I'd be careful about using keywords for this. If you did something like `(= (keyword first-name) :explicitly-nil)` you'd be creating and interning a new keyword. Keywords never get garbage collected. So if this endpoint gets called 10,000 times you could end up with 20,000 keywords sitting around in memory that never get used again. It's probably not a big deal for a create-user endpoint IRL but I just wanted to point out indiscriminate creation of keywords can OOM your instance so you should be careful when doing it

Reply:
The `:explicitly-nil` key is set in the PUT /user/:id endpoint. But your point is taken. I'll see if I can come up with a cleaner approach.
And yea, forgot to add a test here. I'll do that as well.

Reply:
I've reworked this to avoid using a key here at all and am instead passing around a map.